gratipay/models/package/__init__.py

```diff
@@ -40,3 +51,35 @@ def from_names(cls, package_manager, name):
         """
         return cls.db.one("SELECT packages.*::packages FROM packages "
                           "WHERE package_manager=%s and name=%s", (package_manager, name))
+
+
+    @property
+    def team(self):
+        """A computed attribute, the :py:class:`~gratipay.models.team.Team`
+        linked to this package if there is one, otherwise ``None``. Makes a
+        database call.
+        """
+        return self._load_team(self.db)
+
+
+    def ensure_team(self, cursor, owner):
+        """Given a db cursor and :py:class:`Participant`, insert into ``teams`` if need be.
+        """
+        team = self._load_team(cursor)
```

Comment:
If `cursors` is actually `self.db` then yes, there is an implicit transaction under here. However, the purpose of exposing the `cursor` parameter is precisely to allow passing in a proper cursor, so that an entire call chain can operate within one transaction to preserve integrity. In some methods `cursor` is optional ([e.g.](https://github.com/gratipay/gratipay.com/blob/f9a9fccc82c830937d402469aca80906e7e01ffb/gratipay/models/participant/email.py#L205-L213)), but `cursor` is required and front-loaded here because integrity matters.

The transaction of which `ensure_teams` is a part begins [in `finish_email_verification`](https://github.com/gratipay/gratipay.com/blob/f9a9fccc82c830937d402469aca80906e7e01ffb/gratipay/models/participant/email.py#L233). If `finish_email_verification` is called concurrently with the same input, then ... I don't know. I've been working most of the day on trying to sort this out. What I have so far is in 8da4f9cbcd38ad06514a5b805f0579e528b07084.